gene/delete.go

```diff
@@ -0,0 +1,338 @@
+package gene
+
+import (
+	"errors"
+	"github.com/vertgenlab/gonomics/dna"
+)
+
+var (
+	ErrNoStopFound = errors.New("frameshift was unable to find a stop before the end of the gene")
+)
+
+//TODO EffectPrediction
+// Deletion removes bases from the Gene, predicts the effect, and updates the Gene struct to reflect the change.
+// genomeStartPos should be the base directly BEFORE the deleted bases. genomeEndPos should be the base directly AFTER the deleted bases.
```

Comment:
Is it easier if we keep the deletion bounds to our usual left-close right-open, instead of left-open right-open?

Reply:
Yeah this was a holdover from the days when this was intended to take VCFs. I've changed it to left closed right open. This greatly reduced the number of +/-1 shifts so hopefully the coordinates are a bit easier to follow now.